### ISSUE TITLE
Opposite of `pac solution sync`?

### ISSUE THREAD

**Issue by skfd** [microsoft/powerplatform-vscode#531]
`pac solution sync` brings changes from environment into local `.cdsproj`.

How do I push changes that I did in local `.cdsproj` to back selected environment?

In the UI sense, I expect doing simple command like `pac solution syncback`.

**Comment by devkeydet** [comment]
Right now, the equivalent would be `dotnet build` in the folder that includes the `cdsproj` file.  The output will be a solution zip.  Managed, unmanaged, or both depending on how you have your `cdsproj `configured.  You can then run `pac solution import`.

Based on your feedback, it sounds like you want `pac solution syncback` (or some similar verb) to wrap all these steps into a single gesture.  Is my interpretation correct?

**Comment by skfd** [comment]
Exactly! The scenario is that I `git pull` changes from out shared repo and push those changes to env. Same way `sync` pulls those changes. The idea is to have one command to make env have the state that is captured in git.

Full scenario might require more magic and even be git-aware (for example simple solution import will not delete fields, but it's not a big deal, can be a "known limitation").

I think the verbs should be `pac solution push` and `pac solution pull`. Sync is non-directional word, so it implies that it does both.

**Comment by BetimBeja** [comment]
@devkeydet @petrochuk `pac solution sync` is not honoring the Mapping file specified as follows `<SolutionPackageMapFilePath>$(MSBuildThisFileDirectory)WebResourcesMap.xml</SolutionPackageMapFilePath>` 

Can you please add a parameter to specify the mapping file on the first `pac solution sync` execution? maybe `--map-file {PATH}`

I didn't create a new issue since these requests are all related 😉
